### PR TITLE
Promises

### PR DESCRIPTION
**Pull request by codingtwinky**
harry

Using bluebird in server side, it doe make it easier to understand how git process are executed, parsed and handled.

#### **(Ungit plugins that depended on `git` needs to be [updated](https://github.com/FredrikNoren/ungit/pull/666/files#diff-c5d4fadb71ca95ae8675b2fc23a28a63R27) after this is merged)**

### CODE REVIEW

File: source/utils/fs-async.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+var Promise = require('bluebird');
+var fs = Promise.promisifyAll(require("fs"));
+fs.isFileExists = function(file) {
+  return fs.accessAsync(file, fs.F_OK)
```
Comment (campersau) [inline source/utils/fs-async.js]: Node.js 0.10 does not support [`access`](https://github.com/nodejs/node/commit/165b70f146e163b82a09bb869463708516c08cf6). Maybe that's why [a lot of unit tests fail](https://travis-ci.org/FredrikNoren/ungit/jobs/99651029#L1018)?

Comment (Ajedi32) [inline source/utils/fs-async.js]: Just FYI, Node 0.10.x will only supported by the Node.js team until October of this year. Node 0.12.x support will also be dropped in December of this year. (See https://raw.githubusercontent.com/nodejs/LTS/master/schedule.png)

**Comment by campersau** [comment]
There are some indention problems (4 spaces instead of 2).
For example [here](https://github.com/FredrikNoren/ungit/pull/666/files#diff-7cc6d113be128a4ba2e032f8db376ebaR118), [here](https://github.com/FredrikNoren/ungit/pull/666/files#diff-7cc6d113be128a4ba2e032f8db376ebaR246), [here](https://github.com/FredrikNoren/ungit/pull/666/files#diff-e6c012f745dfa4b9168d48f0486955a4R187), [here](https://github.com/FredrikNoren/ungit/pull/666/files#diff-e6c012f745dfa4b9168d48f0486955a4R252) and [here](https://github.com/FredrikNoren/ungit/pull/666/files#diff-e6c012f745dfa4b9168d48f0486955a4R317)

**Comment by codingtwinky** [comment]
Thanks!

Hopefully it passes with using deprecated function

**Pull request merged**